csrc/optimization/layout_inference.cpp

```diff
@@ -0,0 +1,238 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <ir/all_nodes.h>
+#include <ir/utils.h>
+#include <optimization/layout_inference.h>
+
+namespace nvfuser {
+
+namespace {
+
+// move this to util maybe?
+std::vector<int64_t> ascendingAxes(const std::vector<int64_t>& permutation) {
```

Comment:
Code comment on what it does?

Reply:
sorry this was another confusion from switching back and forth. we don't need this util any more. I'll remove it.